src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -18,7 +19,7 @@ public sealed partial class Pipe
 
         private static readonly Action<object> s_signalReaderAwaitable = state => ((Pipe)state).ReaderCancellationRequested();
         private static readonly Action<object> s_signalWriterAwaitable = state => ((Pipe)state).WriterCancellationRequested();
-        private static readonly Action<object> s_invokeCompletionCallbacks = state => ((PipeCompletionCallbacks)state).Execute();
+        private static readonly Action<PipeCompletionCallbacks> s_invokeCompletionCallbacks = state => state.Execute();
```

Comment:
Just a stylistic question:
Does pipelines use static fields like this rather than letting the compiler do the caching for any particular reason? Avoiding the lazy initialization at the call site or something?

Reply:
It’s just more explicit and clear. We started doing this in asp.net a while ago just because it was more obvious what was going on

Reply:
Fair enough.